lib/ForEach.js

```diff
@@ -17,8 +16,13 @@ ForEach.object = function(arrayOfObjects, handler) {
     if (defined(arrayOfObjects)) {
         for (var i = 0; i < arrayOfObjects.length; i++) {
             var object = arrayOfObjects[i];
-            var offset = defaultValue(handler(object, i), 0);
-            i += offset;
+            var returnValue = handler(object, i);
+            if (typeof returnValue === 'number') {
```

Comment:
It is used by the removal functions: [RemoveUnusedProperties.js#L42](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/2.0/lib/RemoveUnusedProperties.js#L42), to make sure that the `ForEach` adjusts the index when elements get removed.